apps/portal/app/buckets/[address]/_components/metadata-panel.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * A component to display object metadata in a more compact format.
+ * Designed to be used alongside the file preview area.
+ */
+import { ChevronDown, ChevronUp } from "lucide-react";
+import { useState } from "react";
+
+import CollapsedStringDisplay from "@recallnet/ui/recall/collapsed-string-display";
+
+import Metric from "@/components/metric";
+import { arrayToDisplay } from "@/lib/convert-matadata";
+
+interface MetadataPanelProps {
+  object: {
+    blobHash: string;
+    recoveryHash: string;
+    size: bigint;
+    expiry: bigint;
+    metadata: readonly {
+      key: string;
+      value: string;
+    }[];
+  };
+  objectSize: {
+    val: string | number;
+    unit: string | undefined;
```

Comment:
This should always be a `string`. We can pass in `objectSize.unit || ""` into here from the parent component. Upstream of there, we need to update the `formatBytes` function to always return a string for `unit`, but that can be done later.

Reply:
okay, got them both simple. i think i interpreted your idea right, but lmk